chain/network/src/peer/peer_actor.rs

```diff
@@ -1086,6 +1102,32 @@ impl PeerActor {
                     "Received routed message from {} to {:?}.",
                     self.peer_info,
                     msg.target);
+                let for_me = self.network_state.message_for_me(&msg.target);
+                if for_me {
+                    metrics::record_routed_msg_metrics(&self.clock, &msg);
+                }
+
+                // Drop duplicated messages routed within DROP_DUPLICATED_MESSAGES_PERIOD ms
+                let key = (msg.author.clone(), msg.target.clone(), msg.signature.clone());
+                let now = self.clock.now();
+                if let Some(&t) = self.routed_message_cache.get(&key) {
+                    if now <= t + DROP_DUPLICATED_MESSAGES_PERIOD {
+                        self.network_state.config.event_sink.push(Event::RoutedMessageDropped);
```

Comment:
do we also have a metric for this? as this should normally never happen, rigth?

Reply:
We send the same message 3 times, so it is bound to happen normally.
We have https://cs.github.com/near/nearcore/blob/d14ef66bb05ce1ebd8cfd63c702e34f8ae0e4eba/chain/network/src/stats/metrics.rs#L296

But both this "drop duplicated" and the metric exist for a long time now and "drop duplicated" for some reason is not included in that metric. Should I add it?

Reply:
yes. Thanks for the explanation.

Reply:
done